src/components/ConnectButton/index.tsx

```diff
@@ -11,24 +12,27 @@ export interface ConnectButtonProps
 }
 
 export const ConnectButton = (props: ConnectButtonProps) => {
-  const { connected, connect, select, provider } = useWallet();
+  const modalState = useWalletModal();
```

Comment:
nit: this can be destructured as `{ setVisible }` and then you can `useCallback` for the `onClick` to avoid rerendering every time `modalState` changes because `visible` has changed.